proxy/src/source.rs

```diff
@@ -31,21 +32,23 @@ pub struct ProjectId {
     pub domain: String,
 }
 
-impl From<radicle_registry_client::ProjectId> for ProjectId {
-    fn from(id: radicle_registry_client::ProjectId) -> Self {
-        Self {
+impl TryFrom<radicle_registry_client::ProjectId> for ProjectId {
+    type Error = &'static str;
+    fn try_from(id: radicle_registry_client::ProjectId) -> Result<Self, Self::Error> {
+        Ok(Self {
```

Comment:
I don't understand why this would need to be a `TryFrom` if it's _always_ going to return `Ok`. This seems like you're adding context, `Result`, that isn't actually needed :thinking: